feature/gribi/otg_tests/weighted_balancing_test/setup_test.go

```diff
@@ -390,3 +431,29 @@ func debugGRIBI(t testing.TB, dut *ondatra.DUTDevice) {
 		t.Log("afts value not present")
 	}
 }
+
+// incrementedMac uses an mac string and increments it by the given i
+func incrementedMac(mac string, i int) (string, error) {
+	macAddr, err := net.ParseMAC(mac)
+	if err != nil {
+		return "", err
+	}
+	convMac := binary.BigEndian.Uint64(append([]byte{0, 0}, macAddr...))
+	convMac = convMac + uint64(i)
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// waitOtgArpEntry ensures that ARP entries are present on the tx otg interface and traffic could be started
+func waitOTGARPEntry(t *testing.T) {
+	ate := ondatra.ATE(t, "ate")
+	ate.OTG().Telemetry().Interface(ateSrcPort+".Eth").Ipv4NeighborAny().LinkLayerAddress().Watch(
+		t, time.Minute, func(val *otgtelemetry.QualifiedString) bool {
```

Comment:
likely time.Minute should be configurable? or?

Reply:
Added the input "timeout"